backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -304,6 +307,10 @@ public boolean performNextOperation(int completedChildCount) {
             return true;
         }
 
+        if (getParameters().getLiveDiskMigrateStage() == LiveDiskMigrateStage.AUTO_GENERATED_SNAPSHOT_REMOVE_END) {
+            lockManager.releaseLock(removeSnapshotLock);
```

Comment:
wouldn't it be better to place it in endSuccessfully/withFailure?

Reply:
Actually, it looks like it will be enough to put it in `endSuccessfully()`, because `endWithFailure()` --> `isConsiderSuccessful()` will return `true` for `AUTO_GENERATED_SNAPSHOT_REMOVE_END`, thus `endSuccessfully()` will be called.

Reply:
right

Reply:
Ack